File: ebakery-backend/src/main/java/pl/damianmrowinski/ebakerybackend/domain/entity/Product.java
package pl.damianmrowinski.ebakerybackend.domain.entity;

import java.math.BigDecimal;
import java.util.Date;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import lombok.Getter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

@Entity
@Table(name = "product")
@Getter
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private String details;

    private Long amount;

    private BigDecimal price;

    private boolean active;

    private String image;

    @CreationTimestamp
    private Date createdDate;

    @UpdateTimestamp
    private Date updatedDate;

    @ManyToOne
    @JoinColumn(name = "category_id", nullable = false)
    private Category category;

}

File: ebakery-frontend/src/app/shared/model/product.ts
export class Product {
  categoryId: number;
  details: string;
  id: string;
  image: string;
  price: number;
  name: string;

  constructor(categoryId: number,
              details: string,
              id: string,
              image: string,
              price: number,
              name: string) {
    this.categoryId = categoryId;
    this.details = details;
    this.id = id;
    this.image = image;
    this.price = price;
    this.name = name;
  }
}
